### ISSUE TITLE
PreRelease ToDo: Lifetable function

### ISSUE THREAD

**Issue by dakni** [ISAAKiel/mortAAR#14]
the default of the function shall be the "modern" approach, i.e. an acv for all ages younger than 5. [attention: this might/will affect the test]

- multvec shall be printed to the output as well; called "nax" bei Kiefitz [provide appropriate reference]
- style the literature in documentation [and perhaps put some flesh on the bones on the roots of the method]
- document the brand new  Lx equations

**Comment by dakni** [comment]
suggestion: change lines 226-229 to

```r
  necdf[, 'multvec'] <- necdf[, 'a'] / 2
  if (acv %>% is.null %>% `!`) {
    necdf[, 'multvec'][1:length(acv)] <- necdf[, 'a'][1:length(acv)] * acv
  }
```

just for me to understand...since necdf is transported as "class-like" thing to the output group we just put the `multvec` as new column into the dataframe

**Comment by nmueller18** [comment]
multvec is the average number of years lived of an individual that died within a specific age class a. For age classes > 5, this value can be approximated by a/2, for younger individuals this is not sufficient as they die before the middle of the class.
Interestingly, this problem is valid even for empiric modern life tables for the first age group (0-1 years) where the age of death is recorded only in years, not in months. To overcome this problem, Keyfitz 1977 introduces what he calls "separation factor". In essence, this is our multvec. In the lifetables of Keyfitz, multvec appears as "nax" (but please mind the notation!).
Contraintuitively, nax is relatively small in developed countries (e. g. p. 15), but large in development countries or historical data (and, supposedly, also in prehistoric times). This means that if young children die at all in developed countries, they do this very young. I have looked through life tables of development countries, and nax for a=0 and a=1-4 hovers around 0.3 and 1.3, thus our approximation (a/3) makes sense.

Because of the notation, I am not really happy with "nax" as name for the column. It seems especially problematic that then the other terms needed also to be preceded by "n". In the UN Model life tables, it appears as "A(x)". This, however, collides with the current name of the population reconstruction.

In conclusion, I would vote for A(x), and change the population reconstruction which is not a standard element of life tables to something like "rel_popx" (similar to the usage of Kokkotidis/Richter 1991).

**Comment by dakni** [comment]
A(x) would also better fit to the current syntax, hence from my side I support the change.

Btw: The text is nice. We should put it in the readme.md as well...as part of the description why the package is useful in the first place. 

Just for understanding: "Contraintuitively" is counterintuitively?

**Comment by nevrome** [comment]
I started to work on this changes in a new branch [Lx_Ax_nax](/ISAAKiel/mortAAR/tree/Lx_Ax_nax).

git pull 
git checkout Lx_Ax_nax
...

**Comment by nevrome** [comment]
@nmueller18 On a completely unrelated note: Maybe it's better to avoid uploading PDFs here. I mean.. this is a public website.

**Comment by nmueller18** [comment]
@nevrome: Ups, you are right, of course. How do I delete the pdf again?
@dakni: "counterintuitively" was meant, thanks!

I have started to write something for the vignette and the paper I have mentioned. It will be rather lengthy, so I would prefer to divide it into different segments and add content to these separately. This will also be helpful to maintain an overview. Is the [wiki ](https://github.com/ISAAKiel/mortAAR/wiki)the place for that?

**Comment by nevrome** [comment]
Concerning the analysis/text/vignette it's maybe best to create an own repository for that. Something like [this](http://svmiller.com/blog/2016/02/svm-r-markdown-manuscript/) or just [this](http://rmarkdown.rstudio.com/) could get you started, if you would like to use RMarkdown. I made very good experiences writing Manuscripts like this. 

IMHO the wiki is the wrong place for text development. I think it's not intended for this and it's also separated from the version control system. By using the wiki, you make yourself completely dependent on Github (proprietary platform) without the advantages of git (version control system).

**Comment by nmueller18** [comment]
@nevrome: At least I have now deleted the link …

Isn't a repository on its own overkill? I had the idea of using parts of the text as snippets where necessary.

**Comment by nevrome** [comment]
Repositories are for free and there's a difference between a tool set and its application. 

But of course you can also start here either with some more vignettes or by creating a playground folder ([which should be added to the .Rbuildignore](https://rdrr.io/cran/devtools/man/use_build_ignore.html)).

**Comment by nmueller18** [comment]
Is it advisable to create a folder within the vignette folder? In that case, I would do that and create a bunch of text files there (one for each subchapter).

**Comment by nevrome** [comment]
It's possible and maybe useful for early drafting, but IMHO not advisable for something that's going to have multiple chapters. 

AFAIK you don't even have to add the files to the .Rbuildignore if they're in a subdirectory of /vignettes. But please check (ctrl + shift + E) as usual before pushing to the master branch. Text development like this is not a common practice in R packages...

**Comment by nevrome** [comment]
OK the branch Lx_Ax_nax is merged and deleted. There's nothing left to do in this issue, correct @nmueller18 ?

I think we can close it and move on.
